sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -1947,6 +1947,31 @@ class Dataset[T] private[sql](
     CollectMetrics(name, (expr +: exprs).map(_.named), logicalPlan)
   }
 
+  /**
+   * Observe (named) metrics through an [[org.apache.spark.sql.Observation]] instance.
```

Comment:
The test failure is likely from Java doc generation from these references. You'd have to wrap it with backticks for the references Java doc build cannot identify in Unidoc.

Reply:
Can you please point me to the error and the command so I can reproduce this locally? Otherwise I wouldn't know if I am fixing the right thing.

Reply:
I think you can reproduce it via `./build/sbt unidoc`.

Reply:
I tried that one, but it succeeds. I am now trying `./dev/run-tests --parallelism 2 --modules "sql" --excluded-tags "org.apache.spark.tags.ExtendedSQLTest"` as in https://github.com/apache/spark/runs/3102419062#step:9:5.

Reply:
hm, feel free to open a PR and leverage GitHub Actions build

Reply:
oh btw some errors from unidoc are fake. I think the real errors are:

```
[error] /home/runner/work/spark/spark/sql/core/target/java/org/apache/spark/sql/Dataset.java:1800:1:  error: unexpected text
[error]    * This is equivalent to calling {@link Dataset.observe(String, Column, Column*)} but does
[error]                                    ^
[error] /home/runner/work/spark/spark/sql/core/target/java/org/apache/spark/sql/Dataset.java:1804:1:  error: reference not found
[error]    * A user can retrieve the metrics by accessing {@link org.apache.spark.sql.Observation.get}.
[error]                                                          ^
[error] /home/runner/work/spark/spark/sql/core/target/java/org/apache/spark/sql/Observation.java:51:1:  error: exception not thrown: java.lang.InterruptedException
[error]    * @throws java.lang.InterruptedException interrupted while waiting
[error]              ^
```

Reply:
you might have to just wrap these three instances with backtics instead of `[[..]]` as a workaround.

Reply:
Done. How is this merged into master, from another PR?

Reply:
Yes, another PR would be cleaner. You can put the link of new PR here, @EnricoMi .

Reply:
Done, thanks!